domain/operation/state/types.go

```diff
@@ -0,0 +1,100 @@
+type operationRow struct {
+	UUID           string     `db:"uuid"`
+	Summary        *string    `db:"summary"`
+	EnqueuedAt     time.Time  `db:"enqueued_at"`
+	StartedAt      *time.Time `db:"started_at"`
+	CompletedAt    *time.Time `db:"completed_at"`
+	Parallel       bool       `db:"parallel"`
+	ExecutionGroup *string    `db:"execution_group"`
+}
+
+// operationTaskRow represents the struct for querying the operation_task table.
+type operationTaskRow struct {
+	UUID          string     `db:"uuid"`
+	OperationUUID string     `db:"operation_uuid"`
+	EnqueuedAt    time.Time  `db:"enqueued_at"`
+	StartedAt     *time.Time `db:"started_at"`
+	CompletedAt   *time.Time `db:"completed_at"`
+}
+
+// operationUnitTaskRow represents the struct for querying operation_unit_task
+// table.
+type operationUnitTaskRow struct {
+	TaskUUID string `db:"task_uuid"`
+	UnitUUID string `db:"unit_uuid"`
+}
+
+// operationMachineTaskRow represents the struct for querying
+// operation_machine_task table.
+type operationMachineTaskRow struct {
+	TaskUUID    string `db:"task_uuid"`
+	MachineUUID string `db:"machine_uuid"`
+}
+
+// operationTaskStatusRow represents the struct for querying
+// operation_task_status table.
+type operationTaskStatusRow struct {
+	TaskUUID  string     `db:"task_uuid"`
+	StatusID  int        `db:"status_id"`
+	Message   *string    `db:"message"`
+	UpdatedAt *time.Time `db:"updated_at"`
+}
+
+// taskResult represents the result of joining operation with its tasks and
+// receivers.
+type taskResult struct {
+	OperationUUID  string         `db:"operation_uuid"`
+	Receiver       sql.NullString `db:"receiver"`
+	Name           sql.NullString `db:"name"`
+	Summary        sql.NullString `db:"summary"`
+	Parallel       bool           `db:"parallel"`
+	ExecutionGroup sql.NullString `db:"execution_group"`
+	EnqueuedAt     time.Time      `db:"enqueued_at"`
+	StartedAt      sql.NullTime   `db:"started_at"`
+	CompletedAt    sql.NullTime   `db:"completed_at"`
+	Status         string         `db:"status"`
+	StatusMessage  sql.NullString `db:"status_message"`
+	StatusValue    sql.NullString `db:"status_value"`
+	OutputPath     sql.NullString `db:"path"`
+}
+
+// uuid represents a simple wrapper for operation UUID queries.
+type uuid struct {
+	UUID string `db:"uuid"`
+}
+
+// taskParameter represents a parameter key-value pair for a task.
+type taskParameter struct {
+	OperationUUID string `db:"operation_uuid"`
+	Key           string `db:"key"`
+	Value         string `db:"value"`
+}
+
+// taskLogEntry represents a log entry from operation_task_log.
+type taskLogEntry struct {
+	TaskUUID  string    `db:"task_uuid"`
+	Content   string    `db:"content"`
+	CreatedAt time.Time `db:"created_at"`
+}
+
+// taskIdent represents a task ID parameter for queries.
+type taskIdent struct {
+	ID string `db:"task_id"`
+}
+
+// taskStatus represents a task status ID for queries on the
+// operation_task_status table.
+type taskStatus struct {
+	StatusID int `db:"status_id"`
```

Comment:
deleteme: The db is the source of truth for the mapping of status values to status ids. None of the structures for interacting with the db should hold the status ID.